### PR TITLE
webapp generic update, with test and recording

### PR DESCRIPTION
**Pull request by williexu**
---
Fixes https://github.com/Azure/azure-cli/issues/892
This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [y ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [y ] Each command and parameter has a meaningful description.
- [y ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-appservice/HISTORY.rst [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 Release History
 ===============
+unreleased
+++++++++++
+* webapp: added generic update; 'az webapp update'
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/HISTORY.rst]: I think `;` here is a grammar error?

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py [hunk 2]
```diff
@@ -449,6 +449,15 @@
             az webapp create -g MyResourceGroup -p MyPlan -n MyUniqueAppName --runtime "node|6.2" --deployment-local-git
 """
 
+helps['webapp update'] = """
+    type: command
+    short-summary: Update a web app.
+    examples:
+        - name: Update the tags of a  web app.
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: there are one extra white space between `a` and 'web'

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 3]
```diff
@@ -98,6 +98,12 @@ def show_webapp(resource_group_name, name, slot=None, app_instance=None):
     return webapp
 
 
+def update_webapp(instance, client_affinity_enabled=None):
+    if client_affinity_enabled is not None:
```
Comment (troydai) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: `if client_affinity_enabled:`

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 4]
```diff
@@ -98,6 +98,12 @@ def show_webapp(resource_group_name, name, slot=None, app_instance=None):
...
+        instance.client_affinity_enabled = client_affinity_enabled == 'true'
```
Comment (troydai) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: Actually, simply this statement is enough. You don't need to test the condition since the `None == 'true'` is always `False`

Comment (williexu) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: We only want to change the field if it is given though. If it isn't provided, this will always change it to false.

Comment (troydai) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: Ok.
